source/UF/ConnectedTypes.lagda

```diff
@@ -210,3 +254,182 @@ the identity type at one level below. We will assume univalence only when necess
+                                            → is-equiv (consts X Y)
+ constants-map-from-truncated-type-is-equiv Y X-conn Y-trunc =
+  ⌜⌝-is-equiv (maps-from-connected-type-to-truncated-type-const X-conn Y-trunc)
+
+ connected-if-consts-is-equiv : {X : 𝓤 ̇} {n : ℕ₋₂}
+                              → ({𝓥 : Universe} (Y : 𝓥 ̇)
+                               → Y is n truncated
+                               → is-equiv (consts X Y))
+                              → X is n connected
+ connected-if-consts-is-equiv {_} {X} {n} is-equiv-from-trunc = (c , G)
+  where
+   s : (X → ∥ X ∥[ n ]) → ∥ X ∥[ n ]
+   s = section-of (consts X ∥ X ∥[ n ])
+        (equivs-have-sections (consts X ∥ X ∥[ n ])
+        (is-equiv-from-trunc ∥ X ∥[ n ] ∥∥ₙ-is-truncated))
+   H : (consts X ∥ X ∥[ n ]) ∘ s ∼ id
+   H = section-equation (consts X ∥ X ∥[ n ])
+        (equivs-have-sections (consts X ∥ X ∥[ n ])
+         (is-equiv-from-trunc ∥ X ∥[ n ] ∥∥ₙ-is-truncated))
+   c : ∥ X ∥[ n ]
+   c = s ∣_∣[ n ]
+   H' : (consts X ∥ X ∥[ n ]) c ＝ ∣_∣[ n ]
+   H' = H ∣_∣[ n ]
+   G : (v : ∥ X ∥[ n ]) → c ＝ v
+   G = ∥∥ₙ-ind (λ - → truncation-levels-are-upper-closed ∥∥ₙ-is-truncated c -)
+               (λ x → happly H' x)
+
+\end{code}
+
+We will now prove a general result from the HoTT book that characterizes when
+a map is connected (see Lemma 7.5.7.)
+
+\begin{code}
+
+ dependent-equiv-from-truncated-fam-connected-map
+  : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y}
+    {P : Y → 𝓦 ̇} {n : ℕ₋₂} 
+  → ((y : Y) → (P y) is n truncated)
+  → f is n connected-map
+  → ((y : Y) → P y) ≃ ((x : X) → P (f x))
+ dependent-equiv-from-truncated-fam-connected-map
+  {_} {_} {_} {X} {Y} {f} {P} {n}
+  P-trunc f-conn = e
+  where
+   e : ((y : Y) → P y) ≃ ((x : X) → P (f x))
+   e = ((y : Y) → P y)                                         ≃⟨ I ⟩
+       ((y : Y) → (fiber f y → P y))                           ≃⟨ II ⟩
+       ((y : Y) → (x : X) → (p : f x ＝ y) → P y)              ≃⟨ Π-flip' ⟩
+       ((x : X) → (y : Y) → (p : f x ＝ y) → P y)              ≃⟨ III ⟩
+       ((x : X) → P (f x))                                     ■
+    where
+     I = Π-cong fe fe (λ - → maps-from-connected-type-to-truncated-type-const
+                       (f-conn -) (P-trunc -))
+     II = Π-cong fe fe (λ - → curry-uncurry' fe fe)
+     III = Π-cong fe fe (λ - → ≃-sym (Yoneda-equivalence fe' (f -) P))
+   observation : ⌜ e ⌝ ＝ dprecomp P f
+   observation = refl
+
+ dep-precomp-from-truncated-family-is-equiv
+  : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y} {P : Y → 𝓦 ̇} {n : ℕ₋₂} 
+  → ((y : Y) → (P y) is n truncated)
+  → f is n connected-map
+  → is-equiv (dprecomp P f)
+ dep-precomp-from-truncated-family-is-equiv P-trunc f-conn =
+  ⌜⌝-is-equiv (dependent-equiv-from-truncated-fam-connected-map P-trunc f-conn)
+
+ dep-precomp-has-section-if-is-equiv
+  : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y} {P : Y → 𝓦 ̇} 
+  → is-equiv (dprecomp P f)
+  → has-section (dprecomp P f)
+ dep-precomp-has-section-if-is-equiv {_} {_} {_} {_} {_} {f} {P} =
```

Comment:
It looks like `f` and `P` should be explicit arguments.

Reply:
I have no problem making them explicit, but can you elaborate on why 'it looks like' they should be?

Reply:
I'd be surprised if Agda could infer them when you use this (which you don't seem to), but it would also get rid of the ugly `{_}`s.

Reply:
Should I make them explicit everywhere or just in this part of the code?

Reply:
I think making this change consistently (i.e. everywhere in the directly surrounding code) would make sense.